test/system/src/test/java/io/pravega/test/system/MultiControllerTest.java

```diff
@@ -98,24 +108,34 @@ public void getControllerInfo() {
         log.info("Controller Service direct URI: {}", controllerURIDirect);
         controllerURIDiscover.set(URI.create("pravega://" + String.join(",", uris)));
         log.info("Controller Service discovery URI: {}", controllerURIDiscover);
-        URI controllerURI = URI.create("tcp://" + String.join(",", uris));
-        log.info("Controller Service direct URI: {}", controllerURI);
-        controllerClient = new ControllerImpl(ControllerImplConfig.builder()
-                                                                  .clientConfig(ClientConfig.builder()
-                                                                                            .controllerURI(controllerURI)
-                                                                                            .build())
-                                                                  .build(),
-                                              EXECUTOR_SERVICE);
+        controllerClientDirect = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(ClientConfig.builder()
+                        .controllerURI(controllerURIDirect.get())
+                        .build())
+                .build(),
+                EXECUTOR_SERVICE);
+        controllerClientDiscover = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(ClientConfig.builder()
+                        .controllerURI(controllerURIDiscover.get())
+                        .build())
+                .build(),
+                EXECUTOR_SERVICE);
+
+        segmentStoreService = Utils.createPravegaSegmentStoreService(zkUris.get(0), controllerURIDirect.get());
```

Comment:
Why do we have `segService` and `segmentStoreService`? Why do we need both? Sounds like we are creating a segment store twice in this test case.

Also, this method is about getting a controller info, so it is unclear why it should be creating a segment store instance here.

Reply:
This was mainly to fetch the running segment store instance and kill it, now that we are using a different service name `multi-controller` , we no longer need this part of the code.